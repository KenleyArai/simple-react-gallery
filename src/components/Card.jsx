import React, { Component } from "react";
import PropTypes from "prop-types";

import Box from "components/Box";
import LazyImg from "components/LazyImg";
import Header from "components/Header";
import Content from "components/Content";
import Button from "containers/Button";

export default class Card extends Component {
  static propTypes = {
    url: PropTypes.string,
    desc: PropTypes.string,
    title: PropTypes.string,
    link: PropTypes.string,
    clickHandler: PropTypes.func,
    social_payload: PropTypes.shape({
      type: PropTypes.oneOf(["github", "linkedin"]).isRequired,
      url: PropTypes.string.isRequired
    })
  };

  render() {
    var { url, desc, title, link, clickHandler, social_payload } = {
      ...this.props
    };
    return (
      <Box>
        <LazyImg srcSet={url} />
        <Box>
          <Header>{title}</Header>
          <Content>{desc}</Content>
          <Box>
            <Button clickHandler={clickHandler} />
          </Box>
        </Box>
      </Box>
    );
  }
}
