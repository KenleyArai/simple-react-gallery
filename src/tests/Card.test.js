import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
import "jest-styled-components";
configure({ adapter: new Adapter() });

import Card from "components/Card";

const fake_handler = () => {};

const social_payload = {
  type: "github",
  url: "https://github.com"
};

const all_props_payload = {
  url: "https://kenley.nyc3.digitaloceanspaces.com/images/serial-face/0.jpg",
  desc:
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer nec odio. Praesent libero. Sed cursus ante dapibus diam. Sed nisi. Nulla quis sem at nibh elementum imperdiet. Duis sagittis ipsum. Praesent mauris. Fusce nec tellus sed augue semper porta. Mauris massa. Vestibulum lacinia arcu eget nulla. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Curabitur sodales ligula in libero. Sed dignissim lacinia nunc. Curabitur tortor. Pellentesque nibh. Aenean quam. In scelerisque sem at dolor. Maecenas mattis. Sed convallis tristique sem. Proin ut ligula vel nunc egestas porttitor. Morbi lectus risus, iaculis vel, suscipit quis, luctus non, massa. Fusce ac turpis quis ligula lacinia aliquet. Mauris ipsum. Nulla metus metus, ullamcorper vel, tincidunt sed, euismod in, nibh. Quisque volutpat condimentum velit. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Nam nec ante. Sed lacinia, urna non tincidunt mattis, tortor neque adipiscing diam, a cursus ipsum ante quis turpis. Nulla facilisi. Ut fringilla. Suspendisse potenti. Nunc feugiat mi a tellus consequat imperdiet. Vestibulum sapien. Proin quam. Etiam ultrices. Suspendisse in justo eu magna luctus suscipit. Sed lectus. Integer euismod lacus luctus magna. Quisque cursus, metus vitae pharetra auctor, sem massa mattis sem, at interdum magna augue eget diam. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Morbi lacinia molestie dui. Praesent blandit dolor. Sed non quam. In vel mi sit amet augue congue elementum. Morbi in ipsum sit amet pede facilisis laoreet. Donec lacus nunc, viverra nec.",
  title: "Lorem ipsum dolor sit amet, consectetu.",
  link: "https://github.com",
  clickHandler: fake_handler,
  social_payload: social_payload
};

const no_desc_payload = {
  url: "https://kenley.nyc3.digitaloceanspaces.com/images/serial-face/0.jpg",
  title: "Lorem ipsum dolor sit amet, consectetu.",
  link: "https://github.com",
  clickHandler: fake_handler,
  social_payload: social_payload
};

const no_title_payload = {
  url: "https://kenley.nyc3.digitaloceanspaces.com/images/serial-face/0.jpg",
  desc:
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer nec odio. Praesent libero. Sed cursus ante dapibus diam. Sed nisi. Nulla quis sem at nibh elementum imperdiet. Duis sagittis ipsum. Praesent mauris. Fusce nec tellus sed augue semper porta. Mauris massa. Vestibulum lacinia arcu eget nulla. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Curabitur sodales ligula in libero. Sed dignissim lacinia nunc. Curabitur tortor. Pellentesque nibh. Aenean quam. In scelerisque sem at dolor. Maecenas mattis. Sed convallis tristique sem. Proin ut ligula vel nunc egestas porttitor. Morbi lectus risus, iaculis vel, suscipit quis, luctus non, massa. Fusce ac turpis quis ligula lacinia aliquet. Mauris ipsum. Nulla metus metus, ullamcorper vel, tincidunt sed, euismod in, nibh. Quisque volutpat condimentum velit. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Nam nec ante. Sed lacinia, urna non tincidunt mattis, tortor neque adipiscing diam, a cursus ipsum ante quis turpis. Nulla facilisi. Ut fringilla. Suspendisse potenti. Nunc feugiat mi a tellus consequat imperdiet. Vestibulum sapien. Proin quam. Etiam ultrices. Suspendisse in justo eu magna luctus suscipit. Sed lectus. Integer euismod lacus luctus magna. Quisque cursus, metus vitae pharetra auctor, sem massa mattis sem, at interdum magna augue eget diam. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Morbi lacinia molestie dui. Praesent blandit dolor. Sed non quam. In vel mi sit amet augue congue elementum. Morbi in ipsum sit amet pede facilisis laoreet. Donec lacus nunc, viverra nec.",
  link: "https://github.com",
  clickHandler: fake_handler
};

describe("Card", () => {
  it("should render without props", () => {
    const wrapper = shallow(<Card />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render with all props", () => {
    const wrapper = shallow(<Card {...all_props_payload} />);
    expect(wrapper).toMatchSnapshot();
  });
  it("should render without desc prop", () => {
    const wrapper = shallow(<Card {...no_desc_payload} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render without title", () => {
    const wrapper = shallow(<Card {...no_title_payload} />);
    expect(wrapper).toMatchSnapshot();
  });
});
