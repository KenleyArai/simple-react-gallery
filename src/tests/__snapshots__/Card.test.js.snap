// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render with all props 1`] = `
<styled.div>
  <LazyImg
    src=""
    srcSet="https://kenley.nyc3.digitaloceanspaces.com/images/serial-face/0.jpg"
    theme={
      Object {
        "primary": "",
        "secondary": "",
        "tertiary": "",
      }
    }
  />
  <Header>
    Lorem ipsum dolor sit amet, consectetu.
  </Header>
</styled.div>
`;

exports[`Card should render without desc prop 1`] = `
<styled.div>
  <LazyImg
    src=""
    srcSet="https://kenley.nyc3.digitaloceanspaces.com/images/serial-face/0.jpg"
    theme={
      Object {
        "primary": "",
        "secondary": "",
        "tertiary": "",
      }
    }
  />
  <Header>
    Lorem ipsum dolor sit amet, consectetu.
  </Header>
</styled.div>
`;

exports[`Card should render without props 1`] = `
<styled.div>
  <LazyImg
    src=""
    srcSet=""
    theme={
      Object {
        "primary": "",
        "secondary": "",
        "tertiary": "",
      }
    }
  />
  <Header />
</styled.div>
`;

exports[`Card should render without title 1`] = `
<styled.div>
  <LazyImg
    src=""
    srcSet="https://kenley.nyc3.digitaloceanspaces.com/images/serial-face/0.jpg"
    theme={
      Object {
        "primary": "",
        "secondary": "",
        "tertiary": "",
      }
    }
  />
  <Header />
</styled.div>
`;
